### PR TITLE
rustc/rustdoc: Perform name resolver cleanups enabled by #94857

### PR DESCRIPTION
**Pull request by petrochenkov**
Unblocks https://github.com/rust-lang/rust/pull/105462.
r? @oli-obk

### CODE REVIEW

File: compiler/rustc_resolve/src/rustdoc.rs [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+/// Otherwise this function may return anything.
+fn preprocess_link(link: &str) -> String {
+    let link = link.replace('`', "");
+    let link = link.split('#').next().unwrap();
+    let link = link.rsplit('@').next().unwrap();
+    let link = link.strip_suffix("()").unwrap_or(link);
+    let link = link.strip_suffix("{}").unwrap_or(link);
+    let link = link.strip_suffix("[]").unwrap_or(link);
+    let link = if link != "!" { link.strip_suffix("!").unwrap_or(link) } else { link };
+    strip_generics_from_path(&link).unwrap_or_else(|_| link.to_string())
```
Comment (kadiwa4) [inline compiler/rustc_resolve/src/rustdoc.rs:348]: `&link` creates an unnecessary double reference

Comment (petrochenkov) [reply compiler/rustc_resolve/src/rustdoc.rs:348]: Fixed in https://github.com/rust-lang/rust/pull/94857.

**Comment by oli-obk** [comment]
The "regression" is in the `bitmaps` crate. I say "regression" because it's actually more things getting documented. I haven't looked into the details, but I think it now just got more data from dependencies to include in the docs

@bors r+ rollup=never

**Comment by petrochenkov** [comment]
>it's actually more things getting documented ... I think it now just got more data from dependencies to include in the docs

It's not something that should happen with this PR.
I'll try to investigate the `bitmaps` case some time later this week.
It's especially interesting that `bitmaps` is the only regression and other crates are not affected.

**Comment by petrochenkov** [comment]
The regression should be fixed by https://github.com/rust-lang/rust/pull/109399.

**Pull request merged**